Katello/katello
# Fixes #33019 - Don't raise errors when deleting missing Candlepin environment

There are two redmines associated to this issue in order to (hopefully) make cherry-picking back to 6.9 a little less annoying :)


(1) https://github.com/Katello/katello/pull/9270 introduced changes that will cause the Candlepin environment to be recreated on publish or promote if it's missing.

(2) This PR takes that slightly further and prevents a 404 from being raised when a Lifecycle Environment is deleted and for some reason the Candlepin environment is already gone.

A separate downstream PR will be needed to bring (1) back into a z-stream due to lots of refactored code.

**To test**

- Create a lifecycle environment
- Create a new CV, publish and promote while tailing candlepin.log
- See that a new environment was created:
``` 
2021-07-19 14:34:54,898 [thread=http-bio-127.0.0.1-23443-exec-5] [req=0ea37549-8fd7-45f8-836d-9f134a2ecab8, org=, csid=ed493181-9bd2-4805-b916-7d78113075f8] INFO  org.candlepin.servlet.filter.logging.LoggingFilter - Request: verb=POST, uri=/candlepin/owners/Default_Organization/environments
...
2021-07-19 14:34:55,890 [thread=http-bio-127.0.0.1-23443-exec-6] [req=ac55b2c4-d62d-4318-8e04-0a7c8f7e8779, org=, csid=ed493181-9bd2-4805-b916-7d78113075f8] INFO  org.candlepin.servlet.filter.logging.LoggingFilter - Request: verb=GET, uri=/candlepin/environments/e4f4f28fc7d767c47976a34e16c0c107
```

- Delete that environment from the rails console:
` Katello::Resources::Candlepin::Environment.destroy('e4f4f28fc7d767c47976a34e16c0c107')`
- Publish the CV again, and see that the environment is recreated by watching candlepin.log

- Delete the environment again from the Rails console (it should correspond to the LCE you created, not Library)
- Delete the Lifecyle Environment that you created and see that no errors get raised
